index.js

```diff
@@ -292,6 +298,51 @@ const ContainerService = {
       };
     }
     // End-Of-Hack
+
+    Services.obs.addObserver(this, "lightweight-theme-changed", false);
+  },
+
+  registerThemeConnection(api) {
+    // This is only used for theme notifications
+    api.browser.runtime.onConnect.addListener((port) => {
+      this.onThemeChanged((theme, topic) => {
+        port.postMessage({
+          type: topic,
+          theme
+        });
+      });
+    });
+  },
+
+  triggerThemeChanged(theme, topic) {
+    this._onThemeChangedCallbacks.forEach((callback) => {
+      callback(theme, topic);
+    });
+  },
+
+  observe(subject, topic) {
```

Comment:
I don't know much about the observers service. I assume that the `observe` name is somewhat magical? I.e., the `Services.obs.addObserver(this, ...)` call calls `this.observe(subject, topic)` which enters here?